crates/types/src/config/worker.rs

```diff
@@ -400,18 +402,38 @@ pub struct SnapshotsOptions {
     ///
     /// This setting does not influence explicitly requested snapshots triggered using `restatectl`.
     ///
-    /// Default: `None` - automatic snapshots are disabled by default
+    /// Default: `None` - automatic snapshots are disabled
     pub snapshot_interval_num_records: Option<NonZeroU64>,
 
     #[serde(flatten)]
     pub object_store: ObjectStoreOptions,
+
+    /// # Error retry policy
+    #[serde(default = "SnapshotsOptions::default_retry_policy")]
```

Comment:
Are you sure we need to specify this default here and not only in the Default implementation? I think how figment work makes this unnecessary since it merges the Default with the loaded config, hence fields that are not configured will always use the default defined in your Default implementation.

Reply:
Evidently not! I just retested this, and it's working completely fine - I was getting an error from this exact launch with one of the earlier revisions:

```
RESTATE_WORKER__SNAPSHOTS__DESTINATION=s3://restate/snapshots restate-server --dump-config
```

The effective config is:

```
[worker.snapshots]
destination = "s3://restate/snapshots"

[worker.snapshots.object-store-retry-policy]
type = "exponential"
initial-interval = "100ms"
factor = 2.0
max-attempts = 10
max-interval = "10s"
```

I'll take this out, thank you!